src/main/java/com/mraof/minestuck/entity/underling/LichEntity.java

```diff
@@ -44,7 +45,8 @@ protected void registerGoals()
 	{
 		super.registerGoals();
 		this.goalSelector.addGoal(1, new AttackResistanceGoal());
-		this.goalSelector.addGoal(3, new AttackingAnimatedEntity.DelayedAttackGoal(this, 1F, false, 14, 16));
+		this.goalSelector.addGoal(2, new SlowAttackWhenInRangeGoal<>(this, 14, 16));
+		this.goalSelector.addGoal(3, new MoveToTargetGoal(this, 1F, false));
```

Comment:
General goal mechanics (from my own observations):
Goals are like a type of ai behaivor. Each registered goal is either active or not active, and each goal has functions that determine if it can become active, if it can remain active, what happens when the goal activates or deactives, and what happens each tick that the goal is active.
One thing that makes goals special is that each goal can have a set of flags attached. Movement and looking are two good examples. For each flag, there can only be at most one active goal that has that flag. In effect, there can for example only be one active goal that controls movement at the same time. Now, if a goal is ready to be activated, but some other goal with shared flags is already active, that second goal might be interrupted, depending on the registered priorities. A goal with a lower number is able to interrupt a goal with a higher number, but not the other way around.

Description of the lichs goals:
In the vanilla goal, it'd make the entity move towards a target, and when close enough, it'd do an instant attack. Here, I've opted to devide the movement part and the attack part into two different goals, to make it possible to interrupt the movement part without interrupting the attacking part.
As such, we have here MoveToTargetGoal (which has the entity move like any vanilla melee mob, but without the attack part) and SlowAttackInRangeGoal (which does the actual attack). When the second goal activates, the slow attack starts. If it were to be interrupted, the slow attack would be interrupted.

Feel free to ask questions or try to summarize what you understand if you need more specific explanations.

Reply:
I am fine with these explanations, good work!